MapboxCoreNavigation/RouteController.swift

```diff
@@ -355,10 +355,25 @@ open class RouteController: NSObject {
         locationManager.stopUpdatingHeading()
     }
     
+    /**
+     The most recently received user location. Snapped to the route line, if applicable, otherwise raw.
+     - seeAlso: snappedLocation, rawLocation
+     */
+    @objc public var location: CLLocation? {
+        return snappedLocation ?? rawLocation
```

Comment:
It might be best to not fallback here to raw location. It sort of conflates the two keys on the progress update.

Reply:
Replied [here.](https://github.com/mapbox/mapbox-navigation-ios/pull/1188#issuecomment-370519058)

Reply:
I'm not sure this was answered. These values are two very different things and i think it's important that if snapped location is nil, we need to say explicitly it's nil. Would you mind answering inline?

Reply:
@JThramer just flagging the above ^

Reply:
Posterity: Resolved the outstanding questions here via a slack call w/ @bsudekum.